sources/api/schnauzer/src/helpers.rs

```diff
@@ -1301,6 +1304,37 @@ pub fn etc_hosts_entries(
     Ok(())
 }
 
+// This helper checks if any objects have '"enabled": true' in their properties.
+//
+// any_enabled takes one argument that is expected to be an array of objects,
+// then checks if any of these objects have an "enabled" property set to true.
+// If the argument is not an array of objects, does not have an "enabled" property,
+// or does not include any objects with this property set to true, it evaluates
+// to be falsey.
+handlebars_helper!(any_enabled: |arg: Value| {
+    #[derive(Deserialize)]
+    struct EnablableObject {
+        enabled: bool,
+    }
+
+    let mut result = false;
+    match arg {
+        Value::Array(items) => {
+            for item in &items {
+                let res = serde_json::from_value::<EnablableObject>(item.clone());
+                if let Ok(eo) = res {
+                    if eo.enabled {
+                        result = true;
+                        break;
+                    }
+                }
+            }
+        }
+        _ => ()
+    }
+    result
```

Comment:
Not a very important nit, take it or leave it:

```suggestion
    match arg {
        Value::Array(items) => {
            for item in &items {
                let res = serde_json::from_value::<EnablableObject>(item.clone());
                if let Ok(eo) = res {
                    if eo.enabled {
                        return true;
                    }
                }
            }
        }
        _ => ()
    }
    false
```

Reply:
If I need to touch this part again I will update, but if you don't feel strongly I'd like to avoid some rebasing and churn as I try to get everything addressed.

Reply:
Now I remember why I went this way. I tried this out in the new PR for the helper and had some issues. This may be due to my lack of deep understanding of how macros work in Rust, but with the `handlebars_helper` macro we are not able to just `return true;`. That causes a compile time error with it expecting the returned type to be a `Result()`.

I'm not sure why we are able to return a boolean with the last statement just being `result` but can't explicitly return a bool. But the helper macro takes that and creates the correct return value.

So to directly return from within these loops, we would need to import a couple more structs and change the lines to:

```rust
use handlebars::{JsonValue, ScopedJson};
...
return Ok(ScopedJson::Derived(JsonValue::from(true)))
```

It seems more readable to me to have the current result variable, but if you would rather see it done the other way (or if you can educate me on rust with a better way to accomplish this!) please add that to the new PR for this part: https://github.com/bottlerocket-os/bottlerocket/pull/2541